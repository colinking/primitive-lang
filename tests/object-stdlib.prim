// SYNTAX TEST "source.prim" "object with stdlib types"

namespace airplane

object TestObject123 {
  string String
  int Int
  boolean Boolean
  timestamp Timestamp
  stringArray []String
}
