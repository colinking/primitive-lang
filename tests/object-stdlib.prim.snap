>// SYNTAX TEST "source.prim" "object with stdlib types"
#^^ source.prim comment.line.double-slash.prim punctuation.definition.comment.prim
#  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ source.prim comment.line.double-slash.prim
>
>namespace airplane
#^^^^^^^^^ source.prim keyword.other.namespace.prim
#         ^ source.prim
#          ^^^^^^^^ source.prim entity.name.namespace.prim
>
>object TestObject123 {
#^^^^^^ source.prim meta.object.prim keyword.other.object.prim
#      ^ source.prim meta.object.prim
#       ^^^^^^^^^^^^^ source.prim meta.object.prim entity.name.type.prim
#                    ^ source.prim meta.object.prim
#                     ^ source.prim meta.object.prim punctuation.definition.block.prim
>  string String
#^^ source.prim meta.object.prim
#  ^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.prim
#        ^ source.prim meta.object.prim meta.field.declaration.prim
#         ^^^^^^ source.prim meta.object.prim meta.field.declaration.prim support.type.prim
>  int Int
#^^ source.prim meta.object.prim
#  ^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.prim
#     ^ source.prim meta.object.prim meta.field.declaration.prim
#      ^^^ source.prim meta.object.prim meta.field.declaration.prim support.type.prim
>  boolean Boolean
#^^ source.prim meta.object.prim
#  ^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.prim
#         ^ source.prim meta.object.prim meta.field.declaration.prim
#          ^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim support.type.prim
>  timestamp Timestamp
#^^ source.prim meta.object.prim
#  ^^^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.prim
#           ^ source.prim meta.object.prim meta.field.declaration.prim
#            ^^^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim support.type.prim
>  stringArray []String
#^^ source.prim meta.object.prim
#  ^^^^^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.prim
#             ^ source.prim meta.object.prim meta.field.declaration.prim
#              ^^ source.prim meta.object.prim meta.field.declaration.prim punctuation.definition.array.prim
#                ^^^^^^ source.prim meta.object.prim meta.field.declaration.prim support.type.prim
>  optional []String?
#^^ source.prim meta.object.prim
#  ^^^^^^^^ source.prim meta.object.prim meta.field.declaration.prim entity.name.prim
#          ^ source.prim meta.object.prim meta.field.declaration.prim
#           ^^ source.prim meta.object.prim meta.field.declaration.prim punctuation.definition.array.prim
#             ^^^^^^ source.prim meta.object.prim meta.field.declaration.prim support.type.prim
#                   ^ source.prim meta.object.prim meta.field.declaration.prim punctuation.other.optional.prim
>}
#^ source.prim meta.object.prim punctuation.definition.block.prim
>